### PR TITLE
move getVmStatus to kubevirt plugin

### PR DESCRIPTION
**Pull request by rawagner**
If we apply this PR on top of https://github.com/openshift/console/pull/2232 vendors~main-chunk drops to 4.92 MB

@spadgett @mareklibra 
size

### CODE REVIEW

**Comment by vojtechszocs** [comment]
Agreed with @spadgett on having sensible helpers, from both logic & type signature perspective.

Lots of KubeVirt related code comes from `kubevirt-web-ui-components` which does **not** leverage TypeScript. As a result, lots of functions might disrupt type checking by using e.g. `_.get` method.

So when moving code from `kubevirt-web-ui-components` to Console monorepo, we should ensure that we stick to existing code standards. Adding specialized (more specific than `any` parameter) signatures is ideal, but at the very least, don't return `any` if we know the result has a specific type.

**Comment by rawagner** [comment]
sorry about those missing types :) I went through all functions and added them, I also fixed naming - using `VM`/`VMI` etc.

File: frontend/packages/kubevirt-plugin/src/statuses/pod/pod.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  if (message) {
+    return {
+      status: POD_STATUS_NOT_READY,
+      message,
+    };
+  }
+  return NOT_HANDLED;
+};
+
+const hasOkStatus = (pod: PodKind): Status => {
```
Comment (spadgett) [inline frontend/packages/kubevirt-plugin/src/statuses/pod/pod.ts]: nit: `hasOKStatus`

File: frontend/public/module/k8s/index.ts [hunk 2]
```diff
@@ -247,15 +247,22 @@ export type PodSpec = {
...
 };
 
 type PodPhase = 'Pending' | 'Running' | 'Succeeded' | 'Failed' | 'Unknown';
 
+type ContainerStateValue = {
+  reason?: string;
+  [key: string]: any;
+};
+
 export type ContainerState = {
```
Comment (spadgett) [inline frontend/public/module/k8s/index.ts]: We should probably use https://github.com/openshift/console/blob/master/frontend/public/module/k8s/index.ts#L255-L259

(and update that type if needed)

**Pull request merged**